mlflow/tracing/display/display_handler.py

```diff
@@ -64,29 +143,28 @@ def __init__(self):
             _logger.debug("Failed to register post-run cell display hook", exc_info=True)
 
     def _display_traces_post_run(self, result):
-        if self.disabled:
+        if self.disabled or not validate_environment():
             self.traces_to_display = {}
             return
 
-        # this should do nothing if not in an IPython environment
         try:
-            from IPython import get_ipython
-            from IPython.display import display
-
-            if get_ipython() is None:
-                return
+            from IPython.display import HTML, display
 
             MAX_TRACES_TO_DISPLAY = MLFLOW_MAX_TRACES_TO_DISPLAY_IN_NOTEBOOK.get()
             traces_to_display = list(self.traces_to_display.values())[:MAX_TRACES_TO_DISPLAY]
             if len(traces_to_display) == 0:
                 self.traces_to_display = {}
                 return
 
-            display(
-                self.get_mimebundle(traces_to_display),
-                display_id=True,
-                raw=True,
-            )
+            if is_in_databricks_runtime():
+                display(
+                    self.get_databricks_mimebundle(traces_to_display),
+                    display_id=True,
+                    raw=True,
+                )
+            else:
+                html = HTML(_get_notebook_iframe_html(traces_to_display))
```

Comment:
Q: Can't we pass `traces[0]._repr_mimebundle_()` here like we do for databricks?

Reply:
oh true, i can combine the functions into `get_mimebundle`